File: Anul II/Medii de proiectare si programare/Laborator/lab07/Ubb.BikeContest.Networking/Request/GetParticipantByDataRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ubb.BikeContest.Model;

namespace Ubb.BikeContest.Networking.Request
{
    [Serializable]
    public class GetParticipantByDataRequest : IRequest
    {
        private readonly Participant participant;

        public GetParticipantByDataRequest(Participant participant)
        {
            this.participant = participant;
        }

        public virtual Participant Participant 
        { 
            get { return participant; } 
        }
    }
}

File: Anul II/Medii de proiectare si programare/Laborator/lab11/java/networking/src/main/java/com/ubb/protocol/request/GetParticipantByDataRequest.java
package com.ubb.protocol.request;

import com.ubb.model.Participant;

public class GetParticipantByDataRequest implements Request {
    private final Participant participant;

    public GetParticipantByDataRequest(Participant participant) {
        this.participant = participant;
    }

    public Participant getParticipant() {
        return participant;
    }
}
